solutions/usaco-713.mdx

```diff
@@ -2,87 +2,82 @@
 id: usaco-713
 source: USACO Bronze 2017 February
 title: Why Did the Cow Cross the Road III
-author: Óscar Garries, tickox
+author: Óscar Garries, tickox, Ryan Chou
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_cowqueue_bronze_feb17.html)
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_cowqueue_bronze_feb17.html)
 
-## Solution 1
+We can naively step through all of the cow entries.
+
+## Implementation
+
+**Time Complexity**: $\mathcal{O}(N)$
 
 <LanguageSection>
 <CPPSection>
 
 ```cpp
-#include <bits/stdc++.h>
+#include <iostream>
+#include <cstdio>
+#include <vector>
+#include <algorithm>
 using namespace std;
-typedef long long ll;
-typedef long double ld;
-#define f first
-#define s second
-#define len(x) (int)x.size()
-int main(){
-	ios_base::sync_with_stdio(0); cin.tie(0);
+
+int main() {
 	freopen("cowqueue.in", "r", stdin);
 	freopen("cowqueue.out", "w", stdout);
-	int n;
+	ios_base::sync_with_stdio(0); cin.tie(0);
+	
+	int n, curTime = 0; 
 	cin >> n;
-	int time = 0;
-	pair<int, int> a[n];
-	for (auto &x: a) {
-		cin >> x.f >> x.s;
-	}
-	sort(a, a + n);
-	for (auto x: a) {
-		if (time >= x.f) {
-			time += x.s;
+	
+	vector<pair<int, int>> cows(n);
+
+	for (int i = 0; i < n; i++) cin >> cows[i].first >> cows[i].second;
+	
+	sort(cows.begin(), cows.end(), [](auto &left, auto &right) {
+    	return left.first < right.first;
+	});
+
+	for (int i = 0; i < n; i++) {
+		// this cow was already waiting, add questioning time to curTimerent time.
```

Comment:
what's curTimerent ?

Reply:
oops i did find and replace all because i was changing `cur` to `curTime`